cheroot/errors.py

```diff
@@ -3,6 +3,16 @@
 import errno
 import sys
 
+import cheroot._compat
+
+
+try:
+    from OpenSSL.SSL import SysCallError as _OpenSSL_SysCallError
```

Comment:
Is it possible to handle this in `cheroot.ssl.pyopenssl` and re-raise as `ConnectionError` or similar so that the external code would only deal with the unified exception and wouldn't need to deal with an import scoped to an optional module?

I think this is an abstraction leak and if we can modify methods in TLS readers/writers to handle their specific cases and re-raise as some normalized `ConnectionError`, then the outer layers wouldn't need to know of such implementation details.

It looks like `NoSSLError` and `FatalSSLAlert` exceptions in this module are trying to achieve this. New refactoring should attempt the same.